File: ragflow-automation-rearchitecture/selenium-csharp/Base/BaseTest.cs
using AutomationC_.Utils;
using NUnit.Framework;
using NUnit.Framework.Interfaces;
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;

namespace AutomationC_.Base;

public abstract class BaseTest
{
    protected IWebDriver Driver => WebDriverFactory.Driver;
    protected WebDriverWait Wait => WebDriverFactory.Wait;

    [OneTimeSetUp]
    public void OneTimeSetUp()
    {
        Listeners.TestListener.OnRunStart();
    }

    [OneTimeTearDown]
    public void OneTimeTearDown()
    {
        Listeners.TestListener.OnRunFinish();
    }

    [SetUp]
    public void SetUp()
    {
        var browser = "chrome";
        if (ConfigReader.GetBooleanProperty("BROWSER"))
        {
            browser = ConfigReader.GetProperty("BROWSER");
        }

        WebDriverFactory.GetDriver(browser);

        Listeners.TestListener.OnTestStart(TestContext.CurrentContext.Test.Name);

        Driver.Navigate().GoToUrl(ConfigReader.GetProperty("BASE_URL"));

        var width = int.Parse(ConfigReader.GetProperty(ConstantUtils.Width));
        var height = int.Parse(ConfigReader.GetProperty(ConstantUtils.Height));
        Driver.Manage().Window.Size = new System.Drawing.Size(width, height);
    }

    [TearDown]
    public void TearDown()
    {
        Listeners.TestListener.OnTestFinish(TestContext.CurrentContext);

        WebDriverFactory.CloseBrowser();
    }
}

File: ragflow-automation-rearchitecture/selenium-testng-java/src/main/java/org/zemoso/base/BaseTest.java
package org.zemoso.base;


import org.openqa.selenium.Dimension;
import org.openqa.selenium.WebDriver;
import org.testng.annotations.AfterSuite;
import org.testng.annotations.BeforeSuite;
import org.zemoso.utils.ConstantUtils;

import java.time.Duration;

import static org.zemoso.utils.ConstantUtils.*;

public class BaseTest {
    protected static WebDriver driver;

    @BeforeSuite
    public void setUp() {
        // Initialize driver based on browser from config
        String browser = "chrome";
        if(ConfigReader.getBooleanProperty("BROWSER"))
            browser = ConfigReader.getProperty("BROWSER");
        driver = WebDriverFactory.getDriver(browser);

        // Set implicit wait from config
        long implicitWait = Long.parseLong(ConfigReader.getProperty(IMPLICIT_WAIT_IN_SEC));
        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(implicitWait));

        // Open base URL from config
        driver.get(ConfigReader.getProperty(BASE_URL));

        // window size
        int width = Integer.parseInt(ConfigReader.getProperty(ConstantUtils.WIDTH));
        int height = Integer.parseInt(ConfigReader.getProperty(ConstantUtils.HEIGHT));
        Dimension screenSize = new Dimension(width, height);
        driver.manage().window().setSize(screenSize);
    }

    @AfterSuite
    public void tearDown() {
        WebDriverFactory.closeBrowser();
    }
}
